📜 Add documentation for create eth-app (#618)
TrueFiEng/useDApp: packages/docs/docs/alternative-getting-started.mdx

---

> sidebar_position: 9

Reviewer: Should be below the first `getting started`

> id: alternative-getting-started

Reviewer: ```suggestion
id: create-eth-app
```

Is the `id ` used or needed?

Author: It is just my style vision:)

---

> #  Getting started with create eth-app

Reviewer: ```suggestion
#  Getting started with Create Eth App
```

To quick start development with useDapp run:

```bash yarn
yarn create eth-app my-eth-app
```

It creates folder my-eth-app and inside you have everything created to start.

```
my-eth-app
├── README.md
├── node_modules
├── package.json
├── .gitignore
└── packages
    ├── contracts
    │   ├── README.json
    │   ├── package.json
    │   └── src
    │       ├── abis
    │       │   ├── erc20.json
    │       │   └── ownable.json
    │       ├── addresses.js
    │       └── index.js
    ├── react-app
    │   ├── README.md
    │   ├── package.json
    │   ├── node_modules
    │   ├── public
    │   │   ├── favicon.ico
    │   │   ├── index.html
    │   │   └── manifest.json
    │   └── src
    │       ├── App.css
    │       ├── App.js
    │       ├── App.test.js
    │       ├── ethereumLogo.svg
    │       ├── index.css
    │       ├── index.js
    │       ├── serviceWorker.js
    │       └── setupTests.js
    └── subgraph
        ├── README.md
        ├── abis
        │   └── erc20.json
        ├── package.json
        ├── schema.graphql
        ├── src
        │   └── mappings
        │       ├── tokens.ts
        │       └── transfers.ts
        └── subgraph.yaml
```

## Start

> Just run

Reviewer: ```suggestion
Just run:
```

```bash yarn
yarn react-app:start
```

> After that the app works, so open [http://localhost:3000/](http://localhost:3000/).

Reviewer: ```suggestion
After that the app starts locally, open [http://localhost:3000/](http://localhost:3000/) to view it in the browser.
```

> ## Frameworks

Reviewer: UseDapp doesn't work with any other framework than the default React, so I propose we remove this section from our doc.

Author: Fine

* Not Reinventing The Wheel: Under the hood, you use Create React App, one of the most popular and battle-tested frontend development environments.

## What's included

Your environment will have everything you need to build a modern Ethereum-powered single-page React app:

* Smooth project management via Yarn Workspaces

* Everything included with Create React App: React, JSX, ES6, TypeScript and Flow syntax support

* Template subgraph that indexes the events emitted by an ERC-20 contract

* Minimalist structure for managing the smart contract ABIs and addresses

* Hassle-free updates for the above tools with a single dependency

## Pros

Create Eth App is a great fit for:

* Learning how to write Ethereum-powered apps in a comfortable and feature-rich development environment.
* Starting new Ethereum-powered single-page React applications without wasting time on copy-pasting boilerplates
* Creating examples with React for your Ethereum-related libraries and components.

> ## Code

Reviewer: ```suggestion
## Source code
```

> Source code on Github [here](https://github.com/paulrberg/create-eth-app).

Reviewer: ```suggestion
Source code is available on [Github](https://github.com/paulrberg/create-eth-app).
```
